analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -18,415 +18,427 @@
+                     && ParenthesizedVariableDesignationSyntaxWrapper.IsInstance(leftDeclaration.Designation))
             {
-                var result = 0;
-                if (TupleExpressionSyntaxWrapper.IsInstance(assignmentLeft))
-                {
-                    result = ((TupleExpressionSyntaxWrapper)assignmentLeft).Arguments.Count;
-                }
-                else if (DeclarationExpressionSyntaxWrapper.IsInstance(assignmentLeft)
-                         && (DeclarationExpressionSyntaxWrapper)assignmentLeft is { } leftDeclaration
-                         && ParenthesizedVariableDesignationSyntaxWrapper.IsInstance(leftDeclaration.Designation))
-                {
-                    result = ((ParenthesizedVariableDesignationSyntaxWrapper)leftDeclaration.Designation).Variables.Count;
-                }
-                return result;
+                result = ((ParenthesizedVariableDesignationSyntaxWrapper)leftDeclaration.Designation).Variables.Count;
             }
-
-            static ISymbol FindDeconstructor(IEnumerable<ISymbol> deconstructors, int numberOfArguments) =>
-                deconstructors.FirstOrDefault(m => m.GetParameters().Count() == numberOfArguments && m.DeclaredAccessibility.IsAccessibleOutsideTheType());
+            return result;
         }
 
-        public override void VisitAttribute(AttributeSyntax node)
-        {
-            var symbol = semanticModel.GetSymbolInfo(node).Symbol;
-            if (symbol != null
-                && symbol.ContainingType.Is(KnownType.System_Diagnostics_DebuggerDisplayAttribute)
-                && node.ArgumentList != null)
-            {
-                var arguments = node.ArgumentList.Arguments
-                    .Where(IsValueNameOrType)
-                    .Select(a => semanticModel.GetConstantValue(a.Expression))
-                    .Where(o => o.HasValue)
-                    .Select(o => o.Value)
-                    .OfType<string>();
-
-                DebuggerDisplayValues.UnionWith(arguments);
-            }
-            base.VisitAttribute(node);
-
-            static bool IsValueNameOrType(AttributeArgumentSyntax a) =>
-                a.NameColon == null  // Value
-                || a.NameColon.Name.Identifier.ValueText == "Value"
-                || a.NameColon.Name.Identifier.ValueText == "Name"
-                || a.NameColon.Name.Identifier.ValueText == "Type";
-        }
+        static ISymbol FindDeconstructor(IEnumerable<ISymbol> deconstructors, int numberOfArguments) =>
+            deconstructors.FirstOrDefault(m => m.GetParameters().Count() == numberOfArguments && m.DeclaredAccessibility.IsAccessibleOutsideTheType());
+    }
 
-        public override void VisitIdentifierName(IdentifierNameSyntax node)
+    public override void VisitAttribute(AttributeSyntax node)
+    {
+        if (node.GetName() is "DynamicallyAccessedMembersAttribute" or "DynamicallyAccessedMembers"
```

Comment:
Consider making "DynamicallyAccessedMembers" a const:
```suggestion
        const string DynamicallyAccessedMembers = "DynamicallyAccessedMembers";
        if (node.GetName() is DynamicallyAccessedMembers or $"{DynamicallyAccessedMembers}Attribute"
```